src/combinatorics/simple_graph/basic.lean

```diff
@@ -162,6 +239,27 @@ def degree : ℕ := (G.neighbor_finset v).card
 lemma card_neighbor_set_eq_degree : fintype.card (G.neighbor_set v) = G.degree v :=
 by simp [degree, neighbor_finset]
 
+lemma degree_pos_iff_exists_adj : 0 < G.degree v ↔ ∃ w, G.adj v w :=
+by { simp only [degree, card_pos, finset.nonempty, mem_neighbor_finset] }
+
+instance incidence_set_fintype [decidable_eq V] : fintype (G.incidence_set v) :=
+fintype.of_equiv (G.neighbor_set v) (G.incidence_set_equiv_neighbor_set v).symm
+
+/--
+This is the `finset` version of `incidence_set`.
+-/
+def incidence_finset [decidable_eq V] : finset (sym2 V) := (G.incidence_set v).to_finset
+
+@[simp]
+lemma card_incidence_set_eq_degree [decidable_eq V] :
+  fintype.card (G.incidence_set v) = G.degree v :=
+by { rw fintype.card_congr (G.incidence_set_equiv_neighbor_set v), simp }
+
+@[simp]
+lemma mem_incidence_finset [decidable_eq V] (e : sym2 V) :
+  e ∈ G.incidence_finset v ↔ e ∈ G.incidence_set v :=
+by { dunfold incidence_finset, simp }
```

Comment:
Could this proof be
```suggestion
G.incidence_set.mem_to_finset
```
?

Reply:
That gives me an error (I didn't forget to delete the `by` this time)
```
invalid field notation, type is not of the form (C ...) where C is a constant
  G.incidence_set
has type
  V → set (sym2 V)```

Reply:
Sorry, it should be (G.incidence_set v).

Reply:
Now I get the `too many arguments` error. It seems like just `set.mem_to_finset` worked though